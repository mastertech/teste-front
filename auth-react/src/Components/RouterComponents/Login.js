import React, { useState, useContext } from "react";

import api from '../apis/api'

import { AuthContext } from "../Context/AuthContext";

function Login(props) { 
    const context = useContext(AuthContext)   

    const [state, setState] = useState({ email: "", password: "" });
    
    function handleChange(event) {
        setState({ ...state, [event.currentTarget.name]: event.currentTarget.value });
    }

    async function handleSubmit(event) {
        event.preventDefault();

        try {
        const response = await api.post("/user/login", state);
        // console.log(response);

        context.setLoggedInUser({ ...response.data });
        localStorage.setItem(
            "loggedInUser",
            JSON.stringify({ ...response.data })
        );
        
        props.history.push("/profile");
        } catch (err) {
            console.error(err.response);
        }
    }
        return (
            <div className="container">
                <div className="box-login">
                    <form onSubmit={handleSubmit}>

                        <h1>Login</h1>
                        <div>
                            <label htmlFor="signupFormEmail">E-mail</label>
                            <input
                            type="email"
                            name="email"
                            id="signupFormEmail"
                            value={state.email}
                            onChange={handleChange}
                            /> 
                        </div>

                        <div>
                            <label htmlFor="signupFormPassword">Password</label>
                            <input
                            type="password"
                            name="password"
                            id="signupFormPassword"
                            value={state.password}
                            onChange={handleChange}
                            />
                        </div>
                        <button type="submit">Login!</button>
                    </form>
                </div>

            </div>
        );
    }

export default Login;