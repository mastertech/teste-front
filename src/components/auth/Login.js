import React, { useState,useEffect, useContext } from 'react'
import { useHistory,withRouter } from "react-router-dom"
import { AuthContext } from '../../provider/AuthProvider'
import Loader from '../utils/Loader'

const URL = process.env.REACT_APP_API

function Login() {
    
    const history = useHistory()
    const { setAuthenticated,authenticated } = useContext(AuthContext)
    const [loading,setLoading] =useState(false)
    const [message,setMessage] = useState(null)

    const [data,setData] =useState({
        email:null,password:null
    })

    useEffect(() => {
        console.log(authenticated)
    } ,[])

    const textChange = (e) => {
        const { name, value } = e.target
        setData({...data,[name]:value})
    }

    const submitLogin = async () => {

        const options = {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(data)
        };

        try {
            const response = await fetch(URL, options);
            if(response.status === 200) {
                const user = await response.json()
                setAuthenticated(true)
                history.push('/user',user)
            } 
            
            if(response.status === 404){
                setMessage('Credenciais inválidas')
                return resetMessage(2000)
            }
        } catch (error) {
            setMessage(error)
            return resetMessage(3000)
        }
    }

    const  validateLogin = (e) => {
        e.preventDefault()
        setLoading(true)
        const {password, email} = data

        if(email === null) {
            setMessage('Insira o E-mail para Logar!')
            return resetMessage() 
        }

        if(!validateEmail(email)) {
            setMessage('Formato do E-mail Invalido!')
            return resetMessage()  
        }

        if(password === null) {
            setMessage('Insira a Senha para Logar!')
            return resetMessage()   
        }

        if(password.length < 8) {
            console.log(password.length)
            setMessage('A senha tem que ter no mínimo 8 caracteres')
            return resetMessage()
        }

        setTimeout(() => {
            submitLogin()
        },500)
        
    }

    const  validateEmail = (email) => {
        const response = /\S+@\S+\.\S+/
        return response.test(email)
    }

    const resetMessage = (time=2000) => {
        setLoading(false)
        setTimeout(() => {
            setMessage(null)
        },time)
    }

    return (
        <div className="content">
            <form className="form" method="POST" onSubmit={validateLogin}>
                <header className="title">
                    <h2>Login</h2>
                </header>
                <span className="message">{ message }</span>
                <div>
                    <div className="input-content">
                        <label>E-mail</label>
                        <input 
                            className="input"
                            type="email" 
                            name="email" 
                            placeholder="Digite o E-mail" 
                            onChange={textChange}
                        />
                    </div>
                    <div className="input-content">
                        <label>Senha</label>
                        <input 
                            className="input"
                            type="password" 
                            name="password" 
                            placeholder="Digite sua Senha" 
                            onChange={textChange}
                        />
                    </div>
                    <div className="btn-content">
                        <button 
                        className="btn" 
                        type="submit"
                        onClick={validateLogin}>
                            { loading ? <Loader /> : 'Login' }
                        </button>
                    </div>
                </div>
            </form>
        </div>
    )
}

export default withRouter(Login)