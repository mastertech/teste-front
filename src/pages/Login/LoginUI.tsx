export const LoginUI = () => <h1>Login</h1>