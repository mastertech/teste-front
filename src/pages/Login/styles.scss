.login-container {
  margin: 0 auto;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
}

.login-content {
  padding: 1.5rem;
  width:35rem;
  height: 35rem;
  background-color: #f5f5f5;
  box-shadow: 0 0 15px rgba(0, 0, 0, 0.3);

  form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  h1 {
    text-align: center;
    margin-bottom: 6rem;
    margin-top: 2rem;
  }

  p {
    color:red;
    font-weight: bold;
  }

  button {
    margin-top: 4rem;
  }
}

.login-input-and-label {
  width: 100%;
  display: flex;
  flex-direction: column;
  margin-bottom: 2rem;

  label {
    font-weight: 500;
    color: var(--gray-300);
    margin-bottom: 1rem;
    font-size: 14px;
  }

  input {
    font-weight: 600;
    background-color: inherit;
    padding: 0rem 0rem 1rem 0.5rem;
    border: 0;
    border-bottom: solid 1px #a19c9c;
    color: var(--gray-700);
    outline: none;

    &:focus {
      border-bottom: solid 2px #000;
    }
  }
}