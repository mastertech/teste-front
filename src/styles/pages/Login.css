.container {
  width: 90%;
  max-width: 400px;
  height: fit-content;

  display: flex;
  flex-direction: column;
  align-items: center;

  padding: 3rem 2rem;

  background-color: #FFF;
  -webkit-box-shadow: 0px 5px 15px 0px rgba(0,0,0,0.3); 
  box-shadow: 0px 5px 15px 0px rgba(0,0,0,0.3);

  border-radius: 5px;
}

.container > h1 {
  font-size: 2.2rem;
  color: #333;
}

form {
  width: 100%;

  display: flex;
  flex-direction: column;
  align-items: center;

  margin-top: 3rem;
}

form label {
  width: 100%;

  display: flex;
  flex-direction: column;

  margin-bottom: 2rem;
}

form label span {
  font-size: 1.3rem;
  color: #555;
}

form label input {
  color: #333;
  font-size: 1.6rem;

  border: none;
  border-bottom: 1px solid #999;
  
  margin-top: 1rem;
  padding-bottom: 1rem;

  outline: none;
  
  transition: .5s;
}

form label input:focus {
  border-color: #CDDC39;
}

form button {
  border: none;
  border-radius: 3px;

  margin-top: 5rem;
  padding: 1.5rem 3rem;
  
  background-color: #CDDC39;

  font-weight: 500;

  outline: none;

  transition: .5s;
}

form button:hover {
  filter: brightness(0.9);
}