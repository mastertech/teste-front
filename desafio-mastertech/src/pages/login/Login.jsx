import React from "react";

import { AppContext } from "../../context/AppContext";

import styles from "./Login.module.css";

const Login = () => {
  return <section className={"animeLeft"}>Login</section>;
};

export default Login;
