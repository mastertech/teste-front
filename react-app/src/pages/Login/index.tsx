import { FC, FormEvent, useCallback, useContext, useRef } from 'react'
import { Link } from 'react-router-dom'
import { Button } from '../../components/Button'
import { Card } from '../../components/Card'
import { Input } from '../../components/Input'
import { Title } from '../../components/Title'
import { AuthContext } from '../../context/AuthContext'
import { useSnackBars } from '../../hooks/useSnackBars'
import { isValidEmail } from '../../utils/isValidEmail'
import './styles.scss'

export const Login: FC = () => {
  const { handleLogin, isLoading } = useContext(AuthContext)
  const inputEmailRef = useRef<HTMLInputElement>(null)
  const inputPassword = useRef<HTMLInputElement>(null)

  const { addAlert } = useSnackBars()

  const validateForm = (email: string, password: string) => {
    if (isValidEmail(email) && typeof password === 'string') {
      return true
    }

    return false
  }

  const handleSubmit = useCallback(async (e: FormEvent) => {
    e.preventDefault()

    const email = inputEmailRef.current?.value
    const password = inputPassword.current?.value

    if (!email) {
      return addAlert('Preencha o campo email')
    }

    if (!password) {
      return addAlert('Preencha o campo senha')
    }

    const isValidForm = validateForm(email, password)

    if (!isValidForm) {
      return addAlert('O email digitado não é válido!')
    }

    await handleLogin({ email, password })
  }, [])

  return (
    <form onSubmit={handleSubmit}>
      <Card>
        <Title>Login</Title>
        <Input
          title="Email"
          type="email"
          name="email"
          inputMode="email"
          ref={inputEmailRef}
        />
        <Input
          title="Senha"
          type="password"
          name="password"
          inputMode="text"
          ref={inputPassword}
        />
        <Button type="submit" title="Login ➡️" isLoading={isLoading} />
        <Link to="/any-route" className="easter-egg">
          Clica aqui vai
        </Link>
      </Card>
    </form>
  )
}
