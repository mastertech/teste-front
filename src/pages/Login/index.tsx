import { useState, useContext } from 'react';
import { Redirect, RouteComponentProps } from 'react-router-dom';

import { LoginUI } from './LoginUI';
import { fetchUserData } from '../../adapters/login';
import { UserContext } from '../../context/UserContext';

export const Login = ({ history }: RouteComponentProps) => {
  const { user, login } = useContext(UserContext);

  const [loginData, setLoginData] = useState({ email: '', password: '' });

  const [error, setError] = useState('');

  const handleInput = (event: any) => {
    const input = event.target;
    setLoginData(data => ({ ...data, [input.name]: input.value }))
  }

  const handleLogin = async () => {
    try {
      const user = await fetchUserData(loginData);
      
      login(user, () => {
        history.replace('/');
      });
    } catch (err) {
      setError('Erro inesperado, tente novamente mais tarde.');
    }
  };

  if (user) return <Redirect to="/" />

  return <LoginUI error={error} data={loginData} handleLogin={handleLogin} handleInput={handleInput} />
};