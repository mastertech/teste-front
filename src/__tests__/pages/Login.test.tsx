import React from 'react';
import Login from '../../pages/Login';
import renderer from 'react-test-renderer';

describe('Login page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
