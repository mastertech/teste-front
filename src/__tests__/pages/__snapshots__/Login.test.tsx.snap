// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders correctly 1`] = `
<div
  className="login-container"
>
  <main
    className="login-content"
  >
    <h1>
      Login
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="login-input-and-label"
      >
        <label
          htmlFor="email"
        >
          E-mail
        </label>
        <input
          name="email"
          onChange={[Function]}
          placeholder="mail@example.com"
          type="email"
          value=""
        />
      </div>
      <div
        className="login-input-and-label"
      >
        <label
          htmlFor="password"
        >
          Senha
        </label>
        <input
          name="password"
          onChange={[Function]}
          placeholder="Senha"
          type="password"
          value=""
        />
      </div>
      <p>
        Â 
      </p>
      <button
        className="button"
        type="submit"
      >
        Login
      </button>
    </form>
  </main>
</div>
`;
